modules/samples/producers/jms/src/main/java/org/wso2/carbon/sample/jmsclient/JMSClient.java

```diff
@@ -49,66 +51,130 @@ public static void main(String[] args) {
-                topicConnection = connFactory.createTopicConnection();
-                topicConnection.start();
-                session = topicConnection.createTopicSession(false, Session.AUTO_ACKNOWLEDGE);
-            }else if(broker.equalsIgnoreCase("mb")){
-                properties.load(ClassLoader.getSystemClassLoader().getResourceAsStream("mb.properties"));
-                Context context = new InitialContext(properties);
-                TopicConnectionFactory connFactory = (TopicConnectionFactory) context.lookup("qpidConnectionFactory");
-                topicConnection = connFactory.createTopicConnection();
-                topicConnection.start();
-                session = topicConnection.createTopicSession(false, Session.AUTO_ACKNOWLEDGE);
-            }else if(broker.equalsIgnoreCase("qpid")){
-                properties.load(ClassLoader.getSystemClassLoader().getResourceAsStream("qpid.properties"));
-                Context context = new InitialContext(properties);
-                TopicConnectionFactory connFactory = (TopicConnectionFactory) context.lookup("qpidConnectionFactory");
-                topicConnection = connFactory.createTopicConnection();
-                topicConnection.start();
-                session = topicConnection.createTopicSession(false, Session.AUTO_ACKNOWLEDGE);
-            }else{
-                log.info("Please enter a valid JMS message broker. (ex: activemq, mb, qpid");
-            }
+            if(topicName != null && !topicName.equalsIgnoreCase("")) {
+
+                filePath = JMSClientUtil.getEventFilePath(sampleNumber, format, topicName, filePath);
+
+                TopicConnection topicConnection;
+                TopicConnectionFactory connFactory = null;
+
+                if (broker.equalsIgnoreCase("activemq")) {
+                    properties.load(ClassLoader.getSystemClassLoader().getResourceAsStream("activemq.properties"));
+                    Context context = new InitialContext(properties);
+                    connFactory = (TopicConnectionFactory) context.lookup("ConnectionFactory");
+                } else if (broker.equalsIgnoreCase("mb")) {
+                    properties.load(ClassLoader.getSystemClassLoader().getResourceAsStream("mb.properties"));
+                    Context context = new InitialContext(properties);
+                    connFactory = (TopicConnectionFactory) context.lookup("qpidConnectionFactory");
+                } else if (broker.equalsIgnoreCase("qpid")) {
+                    properties.load(ClassLoader.getSystemClassLoader().getResourceAsStream("qpid.properties"));
+                    Context context = new InitialContext(properties);
+                    connFactory = (TopicConnectionFactory) context.lookup("qpidConnectionFactory");
+                } else {
+                    log.error("Please enter a valid JMS message broker. (ex: activemq, mb, qpid");
```

Comment:
Thanks @dilini-muthumala  - the changes are done in pull request - https://github.com/wso2/product-cep/pull/158